alis.sh

```diff
@@ -753,6 +762,10 @@ function mkinitcpio_configuration() {
         if [ -n "$LUKS_PASSWORD" ]; then
             HOOKS=${HOOKS//!sd-encrypt/sd-encrypt}
         fi
+    elif [ "$GPT_AUTOMOUNT" == "true" ] && [ -n "$LUKS_PASSWORD" ]; then
+        HOOKS=${HOOKS//!systemd/systemd}
+        HOOKS=${HOOKS//!sd-vconsole/sd-vconsole}
+        HOOKS=${HOOKS//!sd-encrypt/sd-encrypt}
```

Comment:
If BOOTLOADER is not systemd, why systemd hooks area activated?

Reply:
From what I understand from https://www.freedesktop.org/software/systemd/man/systemd-gpt-auto-generator.html, if the root partition is encrypted GPT partition automounting requires systemd init. This then translates to using systemd hooks, see e.g. https://wiki.archlinux.org/title/Mkinitcpio#Common_hooks.

Reply:
One thing is the init system (in alis/arch is always systemd) and other is the bootloader (in alis can be systemd-boot, grub or refind).

So, if GPT partition automounting requires systemd init the condition already meets as systemd init is the init used by default in alis/arch. So to me seems this changes are not necessary?

Seems good to you?

Reply:
Hmm.. from practice I can say that the changes were necessary. I am not sure if we are meaning the same thing when talking about the init system. 

Therefore I try to be more specific:
From what I understand from "When systemd is running in the initrd the / partition may be encrypted in LUKS format as well." (https://www.freedesktop.org/software/systemd/man/systemd-gpt-auto-generator.html), is that systemd needs to be included in the initramfs (https://wiki.archlinux.org/title/Arch_boot_process#initramfs). This, to me means „systemd init“ and need to be used instead of busybox init, as determined by the mkinitcpio hooks (https://wiki.archlinux.org/title/Mkinitcpio#Common_hooks).

Reply:
Systemd init/busybox init are also referred to as early init systems.